### DISCUSSION TITLE
Возможность задавать текущий активный пункт выдачи

### DISCUSSION THREAD

**Discussion by ubzor** [cdek-it/widget#61] [Ideas]
По аналогии с методом `updateLocation()`, хотелось бы иметь возможность делать активным определённый пункт, зная его код, например:
```js
widget.selectOffice(code)
```

**Comment by vermorag** [comment]
Добрый день.  При создании виджета, ему можно передать объект [`selected`](https://github.com/cdek-it/widget/blob/main/dist/cdek-widget.es.d.ts#L211), в котором есть офис (код) и адрес (для режима доставки до офиса и адреса соответственно).
Однако данная возможность пока не является готовой для публичного использования, поэтому ее нет в документации.
В 4 версии виджета, скорее всего, вместо кода ПВЗ, будет использоваться его uuid.

**Reply by ubzor** [reply]
Нас интересует выбор пункта выдачи уже после инициализации виджета. Клиент хочет, чтобы выбор пункта выдачи в вашем виджете был продублирован нашими полями для выбора населённого пункта/пункта выдачи, и нам нужно синхронизировать их между собой (ваш виджет и наши поля). В данный момент мы можем получать из виджета выбранные населённый пункт и пункт выдачи, и устанавливать населённый пункт через `updateLocation()`, не хватает только возможности выбрать конкретный пункт выдачи.

**Reply by vermorag** [reply]
Понял. Тогда точно доработка. Вариант с uuid-ом вам подойдет? И обратите внимание на метод [`selectOffice`](https://github.com/cdek-it/widget/blob/main/dist/cdek-widget.es.d.ts#L144).

**Reply by ubzor** [reply]
Вариант с uuid подойдёт, `selectOffice` работает, офис выбирается, спасибо! Но при этом панель с информацией о выбранном офисе при выборе другого офиса через раз открывается-закрывается, т.е. первый вызов `selectOffice` выбирает нужный офис и открывает панель с информацией о нём справа, второй вызов тоже выбирает нужный офис, но панель с информацией закрывается, и так далее.

И ещё заметил такую вещь - карта стала не во всю высоту виджета, это задаётся где-то в стилях:
```css
height: calc(100% - 3.375rem) !important;
```
как будто-бы раньше этого не было

**Reply by ubzor** [reply]
Ситуацию с `selectOffice` [вынес в отдельное issue](https://github.com/cdek-it/widget/issues/64).

**Comment by neverfan** [comment]
У меня схожий вопрос. Есть ли возможность отобразить на карте выбранный адрес доставки до двери после загрузке виджета? 

Кейс 1:
На сайте имеется отдельное поле с вводом полного адреса с автодополнением (сервис DaData). 
Эти адреса пользователь может сохранять в ЛК, и при выборе одного из своих адресов хотелось бы отобразить на карте через виджет куда будет производиться доставка. 

Кейс 2:
Но пожалуй главная проблема в том, что периодически текстовый адрес не соответствует реальной точке доставки т.к. существует множество сел и коттеджных поселков с одинаковыми названиями (даже в пределах одного региона), как раз после ввода адреса хотелось бы показать через виджет точку реальной доставки для подтверждения.
